code/globalincs/flagset.h

```diff
@@ -114,6 +116,61 @@ class flagset {
 		return *this;
 	}
 
+	size_t to_ubyte_vector(SCP_vector<ubyte>& vector_out)
+	{
+		int counter = 0;		// a counter to determine how far to bitshift
+		ubyte  ubyte_out= 0;	// the holder for our bitshifted flags
+
+		for (size_t i = 0; i < SIZE; i++) {
+
+			// this needs to be at the beginning for push_back after the loop
+			// to work properly.
+			if (counter == UBYTE_SIZE) {
+				vector_out.push_back(ubyte_out);
+				counter = 0;
+				ubyte_out = 0;
+			}
+			
+			// check whether that flag was set
+			// there is probably a more efficient way to do this,
+			// but not unless we can copy the underlying data safely.
+			if (values[i]) {
+				ubyte_out |= 1 << counter; 
+			}	
+
+			// iterate
+			++counter;
+		}
+
+		vector_out.push_back(ubyte_out);
+		
+		return vector_out.size();
+	}
+
+	void set_from_vector(const SCP_vector<ubyte>& vector_in) 
+	{
+		size_t flag = 0; 
+
+		// just iterate through the entries, and set flags as appropriate.
+		for (auto& entry : vector_in) {
+
+			for (uint bitshift = 0; bitshift < sizeof(vector_in.front()); bitshift++){
```

Comment:
err, ``sizeof(vector_in.front())`` will be 1, no? since it's in bytes, not bit. I'd expect this to need to be ``UBYTE_SIZE ``

Reply:
Thanks, I rarely use sizeof